packages/office-ui-fabric-react/src/components/ResizeGroup/ResizeGroup.base.tsx

```diff
@@ -360,12 +430,51 @@ export class ResizeGroupBase extends BaseComponent<IResizeGroupProps, IResizeGro
     this._events.on(window, 'resize', this._async.debounce(this._onResize, RESIZE_DELAY, { leading: true }));
   }
 
+  /**
+   * Normal shallow comparison of props and state except for one optimization.
+   * When the only change in both props and state is that the measureContainer part of the state
+   * changed from true to false, we do not need to re-render.
+   */
+  public shouldComponentUpdate(nextProps: IResizeGroupProps, nextState: IResizeGroupState) {
+    if (nextState.measureContainer === false) {
+      // If there is a state change just because of measureContainer going from true -> false and everything else
+      // is the same as before, we do not need to re-render
+      if (shallowCompare(this.props, nextProps) && shallowCompare({ ...this.state, measureContainer: false }, nextState)) {
+        return false;
+      }
+    }
+    return !shallowCompare(this.props, nextProps) || !shallowCompare(this.state, nextState);
+  }
+
   public componentWillReceiveProps(nextProps: IResizeGroupProps): void {
```

Comment:
Adding all this new logic in componentWillReceive props isn't so great, since it's now deprecated and will be removed in the next major release of React. Since you're the author of this logic and therefore have the best understanding of it, it would be great if you could go ahead and move it to a new "approved" location (`componentDidUpdate` or `getDerivedStateFromProps`) before checking in. Without looking in detail at this code, I'm guessing moving it to `componentDidUpdate` would work best, since `getDerivedStateFromProps` must be static and therefore won't have access to the previous state. (Sorry for pointing this out at the last minute.)

Reply:
Good point, but I am not sure sure how trivial that conversion is. Let me see.

Reply:
@whykushal93 Were you able to do this?

Reply:
@whykushal93 this seems to have fallen off the radar, friendly ping.